py/picca/io.py

```diff
@@ -545,21 +549,30 @@ def read_from_desi(nside,in_dir,thid,ra,dec,zqso,plate,mjd,fid,order):
             iv = iv.sum(axis=0)
             w = iv>0
             fl[w]/=iv[w]
-            d = forest(b_ll,fl,iv,t,ra[wt][0],de[wt][0],ztable[t],exp[wt][0],night[wt][0],fib[wt][0],order)
+            reso_sum = b_reso[wt].sum(axis=0)           
+            reso_in_km_per_s=spectral_resolution_desi(reso_sum,b_ll)
+            diff = b_ll*0.0
+            d  = forest(b_ll,fl,iv,t,ra[wt][0],de[wt][0],ztable[t],exp[wt][0],night[wt][0],fib[wt][0],order,diff,reso_in_km_per_s)
             ### R
             iv = r_iv[wt]
             fl = (iv*r_fl[wt]).sum(axis=0)
             iv = iv.sum(axis=0)
             w = iv>0
             fl[w]/=iv[w]
-            d += forest(r_ll,fl,iv,t,ra[wt][0],de[wt][0],ztable[t],exp[wt][0],night[wt][0],fib[wt][0],order)
-            ### Z
+            reso_sum = r_reso[wt].sum(axis=0)
+            reso_in_km_per_s=spectral_resolution_desi(reso_sum,r_ll)
+            diff = r_ll*0.0
+            d += forest(r_ll,fl,iv,t,ra[wt][0],de[wt][0],ztable[t],exp[wt][0],night[wt][0],fib[wt][0],order,diff,reso_in_km_per_s)
+            ### Z            
             iv = z_iv[wt]
             fl = (iv*z_fl[wt]).sum(axis=0)
             iv = iv.sum(axis=0)
             w = iv>0
             fl[w]/=iv[w]
-            d += forest(z_ll,fl,iv,t,ra[wt][0],de[wt][0],ztable[t],exp[wt][0],night[wt][0],fib[wt][0],order)
+            reso_sum = z_reso[wt].sum(axis=0)
+            reso_in_km_per_s=spectral_resolution_desi(reso_sum,z_ll)
+            diff = z_ll*0.0
```

Comment:
would a `sp.zeros_like(z_ll)` also work

Reply:
@londumas why would you call a function for such a simple calculation? But just for the sake of it, I've timed three options:

    1) diff = z_ll*0
    2) diff = sp.zeros_like(z_ll)
    3) diff = sp.zeros(z_ll.shape)

 3) is fastest, 1) is about twice slower and 2) is the worst, about 4 times slower than 1). I would leave it as is, but up to @Cyeche .

I guess `numpy` provides `zeros_like` as a useful tool when the array can't be multiplied by zero or whether or not it's an array can change at runtime (so `.shape` might not work), and it's slower since it has to figure out what the argument is.